src/Couchbase.Lite.Shared/Document/DataOps.cs

```diff
@@ -167,6 +169,24 @@ internal static object ToNetObject(object value)
             }
         }
 
+        internal static object ToJsonObject(object value)
+        {
+            switch (value) {
+                case null:
+                    return null;
+                case InMemoryDictionary inMem:
+                    return inMem.ToDictionary();
+                case IDictionaryObject roDic:
+                    return roDic.ToDictionary();
```

Comment:
How do Blob objects get handled?

Reply:
When Blob is passed in as value parameter, it will go to case Blob and handle from there.